byterestclient/__init__.py

```diff
@@ -32,14 +32,18 @@ def __init__(self, token=None, endpoint=None, identifier='byterestclient'):
             'User-Agent': '%s:%s' % (socket.getfqdn(), identifier)
         }
 
-    def request(self, method, path, data=None, *args, **kwargs):
+    def request(self, method, path, data=None, referer=None, *args, **kwargs):
         url = self.format_absolute_url(path)
         request_method = getattr(requests, method)
+        headers = self.headers.copy()
+
+        if referer:
+            headers.setdefault('Referer', referer)
```

Comment:
yes, but you can fetch a frame _before_ make_api_call to see what's requesting an api call in the first place